MTGSDKSwift/MagicalCardManager.swift

```diff
@@ -8,74 +8,127 @@
+                completion(Result.error(networkError!))
+            }
+        }
+        
+        guard let url = URLBuilder.buildURLWithParameters(parameters, andConfig: configuration) else {
+            networkError = NetworkError.miscError("fetchSets url build failed")
+            return
+        }
+        
+        mtgAPIService.mtgAPIQuery(url: url) {
+            result in
+            switch result {
+            case .success(let json):
+                let sets = Parser.parseSets(json: json)
+                completion(Result.success(sets))
+            case .error(let error):
+                networkError = error
+            }
+        }
     }
-    private let parser = Parser()
     
     /**
      Fetch JSON returns the raw json data rather than an Array of Card or CardSet. It will return json for sets or cards depending on what you feed it
-     
-     
      - parameter parameters: either [CardSearchParameter] or [SetSearchParameter]
-      - parameter completion: ([String:Any]?, NetworkError?) -> Void
-     
+     - parameter config: MTGSearchConfiguration
+     - parameter completion: (Result<JSONResults>) -> Void
     */
-    public func fetchJSON(_ parameters: [SearchParameter], completion: @escaping JSONCompletionWithError) {
+    public func fetchJSON(_ parameters: [SearchParameter],
+                          _ configuration: MTGSearchConfiguration = MTGSearchConfiguration.defaultConfiguration,
+                          _ completion: @escaping JSONCompletionWithError) {
         
         var networkError: NetworkError? {
             didSet {
-                completion(nil, networkError)
+                completion(Result.error(networkError!))
             }
         }
         
-        guard let url = urlManager.buildURL(parameters: parameters) else {
+        guard let url = URLBuilder.buildURLWithParameters(parameters, andConfig: configuration) else {
             networkError = NetworkError.miscError("fetchJSON url build failed")
             return
         }
         
         mtgAPIService.mtgAPIQuery(url: url) {
-            json, error in
-            if let error = error {
+            result in
+            switch result {
+            case .success:
+                completion(result)
+            case .error(let error):
                 networkError = NetworkError.requestError(error)
                 return
             }
-            completion(json, nil)
         }
-        
     }
     
     
     /**
      Retreives a UIImage based on the imageURL of the Card passed in
-     
-     
      - parameter card: Card
-     - parameter completion: (UIImage?, NetworkError?) -> Void
-     
+     - parameter completion: (Result<UIImage>) -> Void
      */
     
-    
-    public func fetchImageForCard(_ card: Card, completion: @escaping CardImageCompletion) {
+    public func fetchImageForCard(_ card: Card,
+                                  _ completion: @escaping CardImageCompletion) {
```

Comment:
1. This declaration is relatively short, it could probably stay on a single line.
2. I'd recommend the removal of the second `_`